smartsim/control/controller.py

```diff
@@ -273,6 +268,8 @@ def _launch(self, entities, entity_lists, orchestrator):
 
         # create all steps prior to launch
         steps = []
+        entity_lists = manifest.ensembles
+        entities = manifest.models
         if entity_lists:
```

Comment:
I think since the manifest returns empty lists we can just use ``manifest.ensembles`` and ``manifest.models`` directly in the conditional.

Reply:
Yes, we can do this. The only reason I did not was that currently, every time we call `manifest.ensembles` or `manifest.models`, the whole list of deployable is searched and filtered (i.e. the separate lists are not maintained, always recomputed). Therefore, doing the check and then iterating over the lists means computing them twice. Anyhow, given that I don't think the user will create lists of millions of deployables, this shouldn't impact performances.

Reply:
NVM, we can totally skip the conditional, as we are iterating over the resulting list. If the list is empty, the iteration will simply not be run.